first fv test with services works

tomastigera opened the pull request:
## Description

<!-- A few sentences describing the overall goals of the pull request's commits.
Please include
- the type of fix - (e.g. bug fix, new feature, documentation)
- some details on _why_ this PR should be merged
- the details of the testing you've done on it (both manual and automated)
- which components are affected by this PR
- links to issues that this PR addresses
-->

## Related issues/PRs

<!-- If appropriate, include a link to the issue this fixes.
fixes <ISSUE LINK>

If appropriate, add links to any number of PRs documented by this PR
documents <PR LINK>
-->

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

<!-- Writing a release note:
- By default, no release note action is required.
- If you're unsure whether or not your PR needs a note, ask your reviewer for guidance.
- If this PR requires a release note, update the block below to include a concise note describing
  the change and any important impacts this PR may have.
-->

```release-note
TBD
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

tomastigera committed:
[WIP] ipv4 opts test

tomastigera committed:
[BENCH]

tomastigera committed:
WIP

tomastigera committed:
sort of works

tomastigera committed:
parse icmp related

tomastigera committed:
full benchmark

tomastigera committed:
split nat and split off icmp related natting WIP

tomastigera committed:
[BPF] finished icmp

tomastigera committed:
[XDP] compiles

tomastigera committed:
gofmt

tomastigera committed:
[XDP] no ipv4 options

mgleung committed:
Add missing libraries preventing builds

alex-matei committed:
Improve pod startup time

Moving an interface from one namespace to another is time consuming and
requires the interface to be set back to UP again. Improve the current
implementation by creating the host veth directly in the host namespace.

Signed-off-by: Alexandru Matei <alexandru.matei@uipath.com>

tomastigera committed:
[BPF] clasiffy correctly forwarded traffic

Previously, anything that had source the same as the host was considered
as "from host" by the policy engine and global policies with
applyOnForward=false would apply to such traffic. However, for instance
SNATed traffic due to nat-outgoing would be also considered "from host"
eventhough it originated in a workload.

This fixes it by first looking at a SEEN mark which indicates that the
packet was seen by another endpoint and thus is forwarded with the
exception of packets being routed via the bpf nat iface which is used by
the host processes.

tomastigera committed:
[BPF] skip ipv6 parsing tests

tomastigera committed:
[BPF] load calico_tc_skb_icmp_inner_nat progs

skmatti committed:
Remove remaining node update calls in calico-node startup

Followup to https://github.com/projectcalico/calico/pull/7550

Change-Id: I01c8018c9d47401383f05861865ff0c9b44c506e

tomastigera committed:
fix TestAttach

coutinhop committed:
Fix kubernetes archive GPG key url

Retrieving the GPG key from the url https://packages.cloud.google.com/apt/doc/apt-key.gpg is frequently failing with:
W: GPG error: https://packages.cloud.google.com/apt kubernetes-xenial InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY B53DC80D13EDEF05

This has been causing test failures, especially on the cni-plugin
Windows FV tests.

Use https://dl.k8s.io/apt/doc/apt-key.gpg as the url for the key in
order to fix it. See https://github.com/kubernetes/k8s.io/pull/4837 for
more info.

coutinhop committed:
Add 'process/testing/winfv/' to Windows FV semaphore change-watching

tomastigera committed:
[XDP] use bpf_xdp_load_bytes where available (5.18+)

tomastigera committed:
clean up

caseydavenport committed:
Fix release notes location

Fixed the release tool to link to the correct new location for release notes.

Should be picked to v3.26 as well.

caseydavenport committed:
Merge pull request #7718 from projectcalico/casey-fix-release-notes

Fix release notes location

tomastigera committed:
[LIBBPF] fix build

tomastigera committed:
[BPF] remove legacy ipv6

caseydavenport committed:
Merge pull request #7714 from skmatti/master

Remove remaining node update calls in calico-node startup

tomastigera committed:
Merge pull request #7707 from tomastigera/tomas-bpf-fix-gp-on-forward-false

[BPF] clasiffy correctly forwarded traffic

mgleung committed:
Merge pull request #7716 from coutinhop/pedro-CORE-9661

Fix kubernetes archive GPG key url

coutinhop committed:
Fix winfv paths for semaphore change detection

tomastigera committed:
[BPF] IPv6 data plane code passes basic unittests

The BPF dataplane code is no compiled either for 4 byte (ipv4) or 16
byte (ipv6) addresses. Most of the code base is the same for both
versions except IP header parsing.

So code reshiffling and simplification needed due to limitted stack
space.

tomastigera committed:
lock in bpf/mock/map

What is protected by the syscalls in the real workld neeeds to be
protected by a lock when mocking.

mgleung committed:
Merge pull request #7723 from coutinhop/pedro-winfv-semaphore

Fix winfv paths for semaphore change detection in semaphore yaml

tomastigera committed:
[BPF] tc_state_fill_from_iphdr fills ctx->ipheader_len

Then the lenght is stored in state and carried over to subsequent
programs without the need of reading it from the packet itself again.

mgleung committed:
Merge pull request #7726 from mgleung/fix-missing-libs

Add missing libraries preventing builds

nelljerram committed:
Revert GIT_DEPTH setting that now misses non-main branches

(Following devstack commit b0bd5b92, which broke the git clone process when using GIT_DEPTH.)

nelljerram committed:
Merge pull request #7729 from nelljerram/devstack-git

Revert GIT_DEPTH setting that now misses non-main branches

si458 committed:
add podsecurity to operator

si458 committed:
set in generate not yaml

si458 committed:
fix spacing

si458 committed:
add back to original file

caseydavenport committed:
Merge pull request #7705 from UiPath/improve-startup-time

Improve pod startup time

marvin-tigera committed:
Merge pull request #7732 from si458/patch-1

add podsecurity to operator

lwr20 committed:
Add fossa scan to release branches and master

Behnam-Shobiri committed:
Create codeql.yml (#7720)

sridhartigera committed:
Fips changes (#7629)

* Changes to build separate FIPS images

sridhartigera committed:
Fix node-driver-registrar build

mgleung committed:
Merge pull request #7740 from sridhartigera/csi-build-fix

Fix node-driver-registrar build

nelljerram committed:
Uninstall the setuptools that is present on Semaphore nodes

There is loads of stuff pre-installed on a Semaphore ubuntu2004 node, including Python stuff that is:

- distribution-installed (apt-get), in /usr/...

- sudo pip installed, in /usr/local/...

- pip installed in various specific Python virtual environments.

Yesterday there was an update to that - https://docs.semaphoreci.com/reference/semaphore-changelog/
- and that has caused our OpenStack Yoga testing to fail.  Removing the pre-installed /usr/local/
setuptools brings it back to green again.

oxr463 committed:
Fix image pull secret

Whenever more than one image pull secret is specified,
the template renders them all as a single file

marvin-tigera committed:
Merge pull request #7752 from oxr463/fix-tigera-operator-image-pull-secrets

Fix image pull secret for multiple registries

nelljerram committed:
Debugging to explore why Yoga devstack test is bailing early

marvin-tigera committed:
Merge pull request #7747 from nelljerram/devstack-fix

Uninstall the setuptools that is present on Semaphore nodes

nelljerram committed:
Debugging to explore why Yoga devstack test is bailing early

nelljerram committed:
Debugging to explore why Yoga devstack test is bailing early

sridhartigera committed:
Update ciphers for tls 1.2, tls 1.3 for non-fips mode

nelljerram committed:
Debugging to explore why Yoga devstack test is bailing early

nelljerram committed:
Debugging to explore why Yoga devstack test is bailing early

sridhartigera committed:
update ciphers

mgleung committed:
Merge pull request #7756 from sridhartigera/tls-config-change

Update ciphers for tls 1.2, tls 1.3 for non-fips mode

nelljerram committed:
Revert "Debugging to explore why Yoga devstack test is bailing early"

This reverts commit 599681fb23f24e0d8bcc1ba10bdb3b3c85ca5986.

nelljerram committed:
Revert "Debugging to explore why Yoga devstack test is bailing early"

This reverts commit eaab85a89e29d371345e530e07e2620955c56e3f.

nelljerram committed:
Revert "Debugging to explore why Yoga devstack test is bailing early"

This reverts commit 55efd3bf3e11fa31811c03ebc63c7c58dc2a1083.

nelljerram committed:
Revert "Debugging to explore why Yoga devstack test is bailing early"

This reverts commit 96679433b0fb1ba12507068718480f67679fe8a1.

nelljerram committed:
Revert "Debugging to explore why Yoga devstack test is bailing early"

This reverts commit b0bf1521edc87c1fcb673374c60be415d649e743.

nelljerram committed:
Merge remote-tracking branch 'origin/master' into run-yoga-devstack-tests

nelljerram committed:
Work around something closing stdin

MichalFupso committed:
Openshift namespace manifests (#7758)

nelljerram committed:
Merge remote-tracking branch 'origin/master' into run-yoga-devstack-tests

nelljerram committed:
Review markup

marvin-tigera committed:
Merge pull request #7753 from nelljerram/run-yoga-devstack-tests

Run yoga devstack tests

nelljerram committed:
Don't let OpenStack debugging cause a flake

I just saw [a CI run](https://tigera.semaphoreci.com/jobs/8faf9151-768f-49a4-8bee-08210c95c2fd) which failed like this:

```
+ sudo find / -name '*setuptools*'
...
find: ‘/proc/3645’: No such file or directory
```

which is a thing that can be caused by a short lived process at the wrong time.

sridhartigera committed:
Remove high vulnerability cipher (#7764)

nelljerram committed:
Merge pull request #7763 from nelljerram/openstack-fix-tweak

Don't let OpenStack debugging cause a flake

lwr20 committed:
Update .semaphore/semaphore-scheduled-builds.yml

lwr20 committed:
Apply suggestions from code review

lwr20 committed:
Update .semaphore/semaphore.yml.tpl

lwr20 committed:
Merge pull request #7733 from lwr20/lwr-add-fossa

Add fossa scan to release branches and master

nelljerram committed:
OpenStack packaging from release branch to testing PPA

Also remove setting of the CALICO_CHECKOUT variable, which we no longer use
because of Felix and networking-calico both living in the same monorepo as the
packaging logic.

nelljerram committed:
Merge pull request #7771 from nelljerram/openstack-release-branch-packaging

OpenStack packaging from release branch to testing PPA

nelljerram committed:
Merge pull request #7774 from nelljerram/more-openstack-release-branch-packaging-v3.26

OpenStack packaging: support really publishing from any branch

mgleung committed:
Update libs

nelljerram committed:
Merge pull request #7775 from nelljerram/more-openstack-release-branch-packaging

Merge pull request #7774 from nelljerram/more-openstack-release-branc…

mgleung committed:
Fix unsafe integer conversions

mgleung committed:
update k8s version

tomastigera committed:
[BPF] IPv6 policy tests

mgleung committed:
Fix checks for integer conversion overflows

mgleung committed:
remove unused import

mgleung committed:
Merge pull request #7792 from Behnam-Shobiri/update-ma-b

Update libs in main branch

skmatti committed:
Use KUBECONFIG env for kuberenetes client if specified

Change-Id: I0ce80ee726d5e5864752a81a6bad946b89d5a6ec

tomastigera committed:
[BPF] do not check value size in DeleteMapEntry

Passing value size is unnecessary and is incorrectly checked when debug
mode is turned on as at the place of the check we do not know whether
a map is per-cpu or not.

sridhartigera committed:
Fix flexvol registry push (#7786) (#7804)

tomastigera committed:
Merge pull request #7797 from tomastigera/tomas-bpf-map-delete-fix

[BPF] do not check value size in DeleteMapEntry

caseydavenport committed:
Merge pull request #7796 from skmatti/master

Use KUBECONFIG env for kuberenetes client if specified

tmjd committed:
Add Makefile utility to pull operator CRD updates

tomastigera committed:
[BPF] RTCache for ipv4/6

tmjd committed:
Update Operator CRDs

tomastigera committed:
[BPF] create maps for ip v4 or v6

tomastigera committed:
[BPF] v4/6 kube-proxy

tobiasgiese committed:
Fix nilpointer if CALICO_NETWORKING_BACKEND=none

There is a nil pointer in the configureASNumber function if the networking backend is none (i.e. policy-only mode)

Signed-off-by: Tobias Giese <tobias.giese@mercedes-benz.com>

tmjd committed:
gen-manifests update

marvin-tigera committed:
Merge pull request #7821 from mercedes-benz/tobiasgiese/fix-nilpointer-policyonly

Fix nilpointer if CALICO_NETWORKING_BACKEND=none

tobiasgiese committed:
Fix startup unit test for policy only mode

Policy only mode will no longer update the BGP spec.
Thus, the unit test should be fixed as well.

Signed-off-by: Tobias Giese <tobias.giese@mercedes-benz.com>

tmjd committed:
Merge pull request #7818 from tmjd/add-oprtr-crd-update

Add Makefile utility to pull operator CRD updates

tomastigera committed:
[BPF] ipv6 ipsets in policy

ti-afra committed:
Enhance kubebuilder validation in felixconfig crd

What
adding kubebuilder validation for felixconfiguration CRD where-ever possible, mostly adding regex to have kubernetes api validate user input to prevent user errors.

Why
customers have asked for option to apply felix configurations via helm install for calico enterprise. Since apiserver is not up at that time, apiversion crd.projectcalico.org/v1 should be used for that workflow. This means that, creating custom resources is not going through the apiserver and missing the validations that comes with it.

related to https://github.com/tigera/calico-private/pull/6386
jira: EV-3846
userstory: PMREQ-466

fasaxc committed:
Workaround Goland linter bug.

Refactor TypedMap to avoid shadowing the unnamed field's methods.

tomastigera committed:
[FV] add ipv6 options to infra

tomastigera committed:
[BPF] bpf fv test somewhat run

fasaxc committed:
Merge pull request #7853 from fasaxc/untyped-map

Workaround Goland linter bug.

tomastigera committed:
[BPF] fix conntrack for v6

marvin-tigera committed:
Merge pull request #7829 from mercedes-benz/tobiasgiese/fix-ut-node-policy-only

Fix startup unit test for policy only mode

coutinhop committed:
Fix panic in 'calicoctl get nodes'

Fix panic when bgpConfig.Spec.ASNumber was nil in calicoctl/calicoctl/commands/common/printer.go (used by calicoctl/calicoctl/resourcemgr/node.go when running 'calicoctl get nodes', https://github.com/projectcalico/calico/issues/7755 for more info).

Add printer config() test to calicoctl/calicoctl/commands/common/printer_test.go.

Run ginkgo bootstrap in calicoctl/calicoctl/commands/common/ (tests were not being run previously).

Consider $(GINKGO_ARGS) when running the 'ut' target in calicoctl/Makefile.

coutinhop committed:
Merge pull request #7858 from coutinhop/pedro-fix-calicoctl-panic

Fix panic in 'calicoctl get nodes'

tomastigera committed:
Merge pull request #7852 from ti-afra/felixconf-validation

Enhance kubebuilder validation in felixconfig crd

tomastigera committed:
[BPF] passed ipv6 service test

fasaxc committed:
Extract out delta-tracking logic from CachingMap (#7844)

* Add DeltaTracker library for tracking map/set changes vs dataplane.
* Rebase BPF caching map on top of DeltaTracker.

rene-dekker committed:
Strip descriptions from openshift manifests

mgleung committed:
Merge pull request #7871 from rene-dekker/ci-1291

Strip descriptions from openshift manifests

fasaxc committed:
Generic multidict.

fasaxc committed:
Add Pending map views to DeltaTracker. (#7870)

Needed some more utility methods and it made
sense to rework the API for those.

ti-afra committed:
fix kubebuilder tags missing "+"

caseydavenport committed:
Merge pull request #7877 from ti-afra/fix-tag

fix kubebuilder tags missing "+"

fasaxc committed:
Update syscall stub to match updated real version. (#7883)

fasaxc committed:
Add deduplicating syncer callbacks buffer. (#7864)

Add deduplicating syncer callbacks buffer.

The deduplicating buffer implements the syncer callbacks
API on its input, and calls another syncer callback on its output.
In-between it maintains an in-order queue of KV updates and a
tracking map to record what is in the queue.

If an update comes in for a key that already has a value in the queue
then the value in the queue is replaced with the updated KV.

This can cause reordering (which is allowed by the syncer API)
but it ensures that the amount of KVs in flight is bounded to the
total size of the datastore even under substantial overload.
The effect is that the client will periodically "skip ahead"
to the more recent state of the datatstore without seeing
intermediate states.

pasanw committed:
Use slices for pools and blocks in L3RouteResolver

pasanw committed:
Extract parseMacForIPVersion

pasanw committed:
Make IP pools configurable in VXLAN topology

pasanw committed:
Support multiple K8s infras in Felix FVs

pasanw committed:
Introduce TopologyContainers abstraction for FVs

pasanw committed:
Introduce createK8sServiceWithoutKubeProxyArgs

pasanw committed:
Extract VXLAN cluster setup methods

pasanw committed:
Setup IPv6 workloads for VXLAN FV

pasanw committed:
Enable VXLANAlways FV testing

pasanw committed:
Extract some conversion methods

zhangguanzhang committed:
used vars for CALICO_BUILD

pasanw committed:
Explicitly handle lack of old pool in L3RR

ti-afra committed:
update cni_version to v1.1.1-calico+go-1.20.7

mgleung committed:
Merge pull request #7903 from ti-afra/cni-update

update cni_version to v1.1.1-calico+go-1.20.7

fasaxc committed:
Merge pull request #7897 from pasanw/upstream-vxlan-changes

VXLAN and VXLAN FV Refactors

fasaxc commented on felix/bpf-gpl/parsing.h:
```diff
@@ -173,13 +173,20 @@ static CALI_BPF_INLINE int tc_state_fill_from_nexthdr(struct cali_tc_ctx *ctx, b
 		break;
+#ifdef IPVER6
+	case IPPROTO_ICMPV6:
+		/* XXX for now, just allow */
```
We will need to be more specific, there are some scary ICMPv6 messages such as RAs, which let you spoof a router and get host to program routes :scream:

fasaxc commented on felix/bpf/polprog/pol_prog_builder.go:
```diff
@@ -406,20 +403,39 @@ func (p *Builder) writeRecordRuleHit(r Rule, skipLabel string) {
 	// Store the IP address, port and protocol.
 	p.b.Load32(R1, R9, ipOffset)
 	p.b.StoreStack32(R1, keyOffset+ipsKeyAddr)
+	if p.forIPv6 {
```
Should/could we use 64-bit loads/stores for IPv6?

coutinhop committed:
Update renamed windows AMIs

Windows AMIs were renamed from 'Windows_Server-*-English-Core-ContainersLatest' to 'Windows_Server-*-English-Core-Base', update them accordingly.

Also remove mentions to unsupported windows versions (e.g 1903, 1909, 2004, 20H2).

coutinhop committed:
Install docker because AMI doesn't have it anymore

mgleung committed:
Merge pull request #7928 from coutinhop/pedro-fix-win-ami

Update renamed windows AMIs

tomastigera committed:
[BPF] use explicit scratch pointer in state

Makes it always aligned and easy to reach.

tomastigera committed:
[BPF] fix typo

tomastigera committed:
[XDP] scratch comment

tomastigera committed:
fixup ut compilation

tomastigera committed:
[BPF] 64bit load/store for ipset keys in v6 policy progs

sridhartigera committed:
Upgrade libbpf 1.2.2 (#7922)

* Upgrade libbpf 1.2.2

dependabot[bot] committed:
Bump requests from 2.20.1 to 2.31.0 in /hack/postrelease

Bumps [requests](https://github.com/psf/requests) from 2.20.1 to 2.31.0.
- [Release notes](https://github.com/psf/requests/releases)
- [Changelog](https://github.com/psf/requests/blob/main/HISTORY.md)
- [Commits](https://github.com/psf/requests/compare/v2.20.1...v2.31.0)

---
updated-dependencies:
- dependency-name: requests
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

fasaxc committed:
Remove Boxed set, not needed in Go 1.20.

In Go 1.20, interfaces now satisfy the comparable constraint
so there's no need for a second Boxed set impl.

fasaxc committed:
Rev go.mod to v1.20.

sridhartigera committed:
Fix jump map key, value types (#7936)

fasaxc committed:
Merge pull request #7935 from fasaxc/go-1.20

Rev go.mod to v1.20.

fasaxc committed:
Merge pull request #7933 from fasaxc/remove-boxed-set

Remove boxed set.Set

fasaxc committed:
Merge pull request #7932 from fasaxc/generic-multidict

Generic multidict.

fasaxc committed:
Merge pull request #7929 from projectcalico/dependabot/pip/hack/postrelease/requests-2.31.0

Bump requests from 2.20.1 to 2.31.0 in /hack/postrelease

fasaxc committed:
Merge pull request #7902 from zhangguanzhang/zhangguanzhang-patch-1

used vars for CALICO_BUILD

tomastigera committed:
Merge pull request #7701 from tomastigera/tomas-bpf-ipv4-opts-2

[BPF] support for ipv4 options

fasaxc committed:
Fix flake in dedupe buffer test. (#7940)

Wait for status to show up as well as keys.

sridhartigera committed:
Fix build warnings (#7941)

* Fix libbpf.go build warnings

* Add -Werror to cgo cflags

fasaxc committed:
Defer flush of the policy resolver (#7939)

Flushing after every match update causes high CPU usage
for endpoints that are being added/removed with many
matching policies.  Defer the sort and flush until we're
ready to send data downstream.

rodrigorfk committed:
fix: adding timeout to the typha server TLS handshake (#7909)

Commit:
calico-kube-controllers: Run as non-root by default for the s390x image
- create status and profiles folder
- create related files and chown to 999

Signed-off-by: Da Li Liu <liudali@cn.ibm.com>

danudey committed:
Add LOCAL_CRANE option to lib.Makefile

danudey committed:
Merge pull request #7960 from danudey/dfox_enable_local_crane

Add LOCAL_CRANE option to lib.Makefile

marvin-tigera committed:
Merge pull request #7949 from tomastigera/tomas-bpf-ipv6

[BPF] ipv6 dataplane code passes first unittests

Commit:
Dump policies without the assembly and replace with more condensed rule counters output (#7954)

lwr20 committed:
Merge pull request #7955 from liudalibj/s390x-image-fix

calico-kube-controllers: Run as non-root by default for the s390x image

tomastigera committed:
[BPF] split calico_tc_skb_accepted_entrypoint

Because of insufficient stack space and hitting the limit way too often
in debug programs, handling new flows (and creating conntrack entries)
is split off into a separate program.

tomastigera committed:
passes make static-checks

tomastigera committed:
[BPF] v6 ctlb progs for v4 moved to connect_balancer_v46.c

marvin-tigera committed:
Merge pull request #7973 from tomastigera/tomas-bpf-accepted-split

[BPF] split calico_tc_skb_accepted_entrypoint

mgleung committed:
Move TLS handshake to per-connection goroutine

tomastigera committed:
[BPF] CTLBv6 bpf code compiles

tomastigera committed:
[BPF] remove unnecessary load in log filter

tomastigera committed:
[BPF] CLTBv6 works

tomastigera committed:
[BPF] fix calico-bpf -6

tomastigera committed:
fv fix infra for ipv6 host ips

tomastigera committed:
more fixes for ipv6 in fv

fasaxc committed:
Create SECURITY.md (#7995)

Create SECURITY.md

Co-authored-by: Anthony T <25327116+anthonytwh@users.noreply.github.com>

fasaxc committed:
Deflake typha grace period tests.

Debug logging was causing slow updates, which
in turn meant that we were unable to cause enough
back pressure to trigger the feature.

fasaxc committed:
Merge pull request #7993 from fasaxc/ms-typha-fix

Move TLS handshake to per-connection goroutine

tomastigera committed:
[BPF] FIBv6 works

marvin-tigera committed:
Merge pull request #7964 from tomastigera/tomas-bpf-ipv6-proxy-rebase

[BPF] ipv6 kube-proxy
